app/models/user.rb

```diff
@@ -1,4 +1,13 @@
 class User < ApplicationRecord
+  has_many :interests
+  before_save :nil_gender
+
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  def nil_gender
```

Comment:
Consider renaming to `ensure_gender_not_nil`. Also consider overwriting the setter instead of using a before_filter (i.e `def gender=(value)`). Here you can find the idea being discussed: http://stackoverflow.com/questions/10464793/what-is-the-right-way-to-override-a-setter-method-in-ruby-on-rails